timemachine/cpp/src/tibd_exchange_move.cu

```diff
@@ -229,7 +229,7 @@ void TIBDExchangeMove<RealType>::move(
             this->num_target_mols_,
             this->batch_size_,
             this->d_sample_segments_offsets_.data,
-            this->d_log_weights_before_.data,
+            d_src_log_weights_.data,
```

Comment:
does num_target_mols_ need to change? (or is this already equal to # of target mols in src)

Reply:
It does not, that is used as an indicator to say that we have gone beyond the length of the arrays. The actual lengths of the source mols is determined by `d_sample_segments_offsets_`